### PR TITLE
feat: add sonar badge to pipeline

### PR DESCRIPTION
**Pull request by adong**
## Context

To add badges, and include one example of `sonar` to the pipeline, so we can update the sonar dashboard name and uri in the pipeline option page.

## Objective
image

image

image

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: models/pipeline.js [hunk 1]
```diff
@@ -90,7 +90,19 @@ const MODEL = {
                 actions: Joi.array().items(Joi.string())
             })
         )
-        .description('List of subscribed scm urls paired with actions')
+        .description('List of subscribed scm urls paired with actions'),
+
+    sonarDashboardName: Joi.string()
```
Comment (tkyi) [inline models/pipeline.js:95]: do you need to add these to `get` (line 133)?

Comment (jithine) [inline models/pipeline.js:95]: what if we have more integrations in the future ? Do we need a flat structure like this ? ie keep adding more fields for potential integrations like Saucelabs Project UI, Mend UI, Semgrep etc ?

Comment (adong) [reply models/pipeline.js]: Updated

File: models/pipeline.js [hunk 2]
```diff
@@ -11,6 +11,16 @@ const Parameters = require('../config/parameters');
 const mutate = require('../lib/mutate');
 
 const STATES = ['ACTIVE', 'INACTIVE'];
+const BADGE = Joi.object({
```
Comment (tkyi) [inline models/pipeline.js:14]:
```suggestion
const SONAR = Joi.object({
```

Comment (adong) [reply models/pipeline.js:14]: Oh, this is not a typo. This is written for extendibility, so we might be able to add something like in the future.

```
{
    sonar: BADGE,
    saucelabs: BADGE
}
```

File: models/pipeline.js [hunk 3]
```diff
@@ -90,7 +97,14 @@ const MODEL = {
...
             })
         )
-        .description('List of subscribed scm urls paired with actions')
+        .description('List of subscribed scm urls paired with actions'),
+
+    badges: Joi.object({
+        sonar: BADGE
+    })
+        .description('A list of badges that pipline has')
+        .example(`sonar: { name: 'dashboard', uri: 'http://sonar.com/sample1'}`)
```
Comment (tkyi) [inline models/pipeline.js:106]:
```suggestion
        .example(`{ sonar: { name: 'dashboard', uri: 'http://sonar.com/sample1' } }`)
```

**Review by tkyi — APPROVED** [review]

**Pull request merged**